src/Operation/BulkWrite.php

```diff
@@ -413,6 +425,14 @@ private function validateOperations(array $operations, ?DocumentCodec $codec, En
                         throw InvalidArgumentException::expectedDocumentType(sprintf('$operations[%d]["%s"][2]["collation"]', $i, $type), $args[2]['collation']);
                     }
 
+                    if (isset($args[2]['sort']) && ! is_document($args[2]['sort'])) {
+                        throw InvalidArgumentException::expectedDocumentType(sprintf('$operations[%d]["%s"][2]["sort"]', $i, $type), $args[2]['sort']);
+                    }
+
+                    if (isset($args[2]['sort']) && $args[2]['multi']) {
+                        throw new InvalidArgumentException(sprintf('"sort" option cannot be used with $operations[%d]["%s"]', $i, $type));
+                    }
```

Comment:
Noted that the `multi` option cannot `true` for `replaceOne`, that why the validation is only for `updateMany` and `updateOne`.